### PR TITLE
Adds support for multisearch

### LINKED ISSUE #29: [FEATURE] Handle multisearch requests
### Is your feature request related to a problem?
Currently, the plugin only handles requests of type `SearchRequest`. The plugin should also handle `MultiSearchRequest` objects.

### What solution would you like?
Modify the action filter to handle `MultiSearchRequest`s. Handle each `SearchRequest` in the `MultiSearchRequest` in the same way the plugin does now.

### PR DESCRIPTION
**Pull request by jzonthemtn**
### Description
See #29 - Adds support for multisearch

### Issues Resolved
#29

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

To test:

- Build the docker image and start with `docker compose up`.
- Create the ecommerce index: `curl -X PUT http://localhost:9200/ecommerce`
- Run an msearch:

```
curl -s -X GET "http://localhost:9200/_msearch" -H 'Content-Type: application/json' -d'
{ "index": "ecommerce"}
{ "query": { "match_all": {} }, "ext": { "ubi": { "query_id": "11111" } } }
{ "index": "ecommerce"}
{ "query": { "match_all": {} }, "ext": { "ubi": { "query_id": "22222" } } }
'
```

Now look at the `ubi_queries` index and you will see two queries that match the query IDs in our `msearch` request.

```
curl localhost:9200/ubi_queries/_search
```

Response:

```
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 2,
      "relation": "eq"
    },
    "max_score": 1,
    "hits": [
      {
        "_index": "ubi_queries",
        "_id": "Sd6JU5IBPUzlZ11s4psz",
        "_score": 1,
        "_source": {
          "query_response_id": "b226e3db-0797-44bc-a349-fc4722881872",
          "user_query": "",
          "query_id": "22222",
          "query_response_object_ids": [],
          "query": "{\"query\":{\"match_all\":{\"boost\":1.0}},\"ext\":{\"query_id\":\"22222\",\"user_query\":null,\"client_id\":null,\"object_id_field\":null,\"query_attributes\":{}}}",
          "query_attributes": {},
          "client_id": "",
          "timestamp": 1727978398255
        }
      },
      {
        "_index": "ubi_queries",
        "_id": "St6JU5IBPUzlZ11s4psz",
        "_score": 1,
        "_source": {
          "query_response_id": "f73c8b07-34ec-43d6-8554-c62372af3c9f",
          "user_query": "",
          "query_id": "11111",
          "query_response_object_ids": [],
          "query": "{\"query\":{\"match_all\":{\"boost\":1.0}},\"ext\":{\"query_id\":\"11111\",\"user_query\":null,\"client_id\":null,\"object_id_field\":null,\"query_attributes\":{}}}",
          "query_attributes": {},
          "client_id": "",
          "timestamp": 1727978398255
        }
      }
    ]
  }
}
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,7 @@
-FROM opensearchproject/opensearch:2.14.0
+FROM opensearchstaging/opensearch:3.0.0
```
Comment (epugh) [inline Dockerfile:1]: is this kind of the pattern that plugins use?

File: scripts/get-indexed-queries.sh
Comment (epugh) [inline scripts/get-indexed-queries.sh:1]: do we want this in....

File: src/main/java/org/opensearch/ubi/UbiActionFilter.java [hunk 2]
```diff
@@ -104,82 +102,100 @@ public <Request extends ActionRequest, Response extends ActionResponse> void app
...
-                                searchResponse.getClusters(),
-                                queryId
-                        );
+                final String queryId = ubiParameters.getQueryId();
+                final String userQuery = ubiParameters.getUserQuery();
+                final String userId = ubiParameters.getClientId();
+                final String objectIdField = ubiParameters.getObjectIdField();
+                final Map<String, String> queryAttributes = ubiParameters.getQueryAttributes();
 
+                // TODO: Ignore the UBI in ext.
```
Comment (epugh) [inline src/main/java/org/opensearch/ubi/UbiActionFilter.java:152]: What does this mean?   I'm fine with the TODO, but would love a bit more detail about what this means ;-)

**Review by epugh — APPROVED** [review]
LGTM.  a Nit or two but otherwise looks great.

File: src/main/java/org/opensearch/ubi/UbiActionFilter.java (hunk 2 above)
Comment (jzonthemtn) [reply src/main/java/org/opensearch/ubi/UbiActionFilter.java:152]: The thought was to not log the `ubi` part of the query to save space. I will take this out and write an issue.

File: Dockerfile (hunk 1 above)
Comment (jzonthemtn) [reply Dockerfile:1]: Yes, it seems to be.

**Pull request merged**